frontend/lib/api/form/pollingstation/pollingStationUtils.ts

```diff
@@ -0,0 +1,135 @@
+        break;
+      case "political_group_votes":
+        if (index !== undefined) {
+          //TODO: check this index vs number thing, it's very confusing/error prone
+          const sectionKey = `political_group_votes_${index + 1}` as FormSectionID;
+          const section = formState.sections[sectionKey];
+          if (section) {
+            section[target].push(validationResult);
+          }
+        }
+        break;
+      default:
+        formState.unknown[target].push(validationResult);
+        break;
+    }
+  });
+}
+
+export function formSectionComplete(section: FormSection): boolean {
+  if (section.isSaved) {
+    if (section.errors.length === 0 || hasOnlyGlobalValidationResults(section.errors)) {
+      if (section.warnings.length === 0 || section.ignoreWarnings) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+export function hasOnlyGlobalValidationResults(arr: ClientValidationResult[]): boolean {
+  return arr.every((result) => isGlobalValidationResult(result));
+}
+
+//get the next section in the form based on index
+export function getNextSection(
+  formState: FormState,
+  currentSection: FormSection,
+): FormSectionID | null {
+  for (const section of Object.values(formState.sections)) {
+    if (section.index === currentSection.index + 1) {
+      return section.id;
+    }
+  }
+
+  return null;
+}
+
+export function resetFormSectionState(formState: FormState) {
+  Object.values(formState.sections).forEach((section) => {
+    section.errors = [];
+    section.warnings = [];
+    section.isSubmitted = undefined;
+  });
+  formState.unknown.errors = [];
+  formState.unknown.warnings = [];
+}
+
+export function currentFormHasChanges(
+  currentForm: AnyFormReference,
+  values: PollingStationValues,
+): boolean {
+  if (currentForm.type === "recounted") {
+    const valA: RecountedValue = { recounted: values.recounted };
+    const valB = currentForm.getValues();
+    return !deepEqual(valA, valB);
+  }
+
+  if (currentForm.type === "voters_and_votes") {
+    const valA: VotersAndVotesValues = {
+      voters_counts: values.voters_counts,
+      votes_counts: values.votes_counts,
+      voters_recounts: values.voters_recounts,
+    };
+    const valB = currentForm.getValues();
+    return !deepEqual(valA, valB);
+  }
+
+  if (currentForm.type === "differences") {
+    const valA: DifferencesValues = {
+      differences_counts: values.differences_counts,
+    };
+    const valB = currentForm.getValues();
+    return !deepEqual(valA, valB);
+  }
+
+  //political_group_votes
+  const valA = values.political_group_votes.find((pg) => pg.number === currentForm.number);
+  if (valA) {
+    const valB = currentForm.getValues();
+    return !deepEqual(valA, valB);
+  }
+
+  return false;
+}
+
+export function isGlobalValidationResult(validationResult: ValidationResult): boolean {
+  switch (validationResult.code) {
+    case "F401":
```

Comment:
F401 is not a global error, this is validated for each list when you press "Volgende". This just checks if the counts within the list add up to the given total.